### PR TITLE
chore: reduce the number of errors that are thrown

### PR DESCRIPTION
**Pull request by Aenimus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: composition/src/errors/errors.ts [hunk 1]
```diff
@@ -415,6 +451,19 @@ export function unexpectedParentKindErrorMessage(
...
 }
 
+export function unexpectedParentKindForChildError(
+  parentTypeName: string,
+  expectedTypeString: string,
+  actualTypeString: string,
+  childName: string,
+  childTypeString: string,
+): Error {
+  return new Error(
```
Comment (StarpTech) [inline composition/src/errors/errors.ts:461]: For a future version, we should throw concrete error types to distinguish between composition and other errors.

Comment (Aenimus) [reply composition/src/errors/errors.ts:461]: I have a decent plan for a whole error overhaul. Will discuss with you at some point.

File: composition/src/normalization/normalization-factory.ts [hunk 2]
```diff
@@ -732,61 +755,48 @@ export class NormalizationFactory {
...
             break;
           }
-          default:
-            throw new Error(`Unknown argument ${arg.name.value} found in event directive`);
         }
       }
 
       if (!topic) {
-        throw new Error(`Event directives must have a topic argument`);
+        return;
```
[the remark this answers is not in the document]
Comment (Aenimus) [reply composition/src/normalization/normalization-factory.ts:799]: It's impossible for topic not to exist. THere is a comment that validation happens elsewhere. Check the tests in events.test.ts.

File: composition/src/schema-building/utils.ts [hunk 3]
```diff
@@ -227,6 +227,14 @@ export function getDefinedArgumentsForDirective(
     // }
     definedArguments.add(argumentName);
   }
+  if (duplicateArgumentNames.size > 0) {
+    errorMessages.push(
+      duplicateDirectiveArgumentDefinitionsErrorMessage(directiveName, hostPath, [...duplicateArgumentNames]),
+    );
+  }
+  if (unexpectedArgumentNames.size > 0) {
+    errorMessages.push(unexpectedDirectiveArgumentErrorMessage(directiveName, [...unexpectedArgumentNames]));
```
[the remark this answers is not in the document]
Comment (Aenimus) [reply composition/src/schema-building/utils.ts:236]: It's always propagated as an error. Related errors are passed into a function that produces the containing error.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**